bottles/backend/managers/manager.py

```diff
@@ -268,37 +262,52 @@ def check_app_dirs(self):
 
     def organize_components(self):
         """Get components catalog and organizes into supported_ lists."""
-        catalog = self.component_manager.fetch_catalog()
-        if len(catalog) == 0:
-            logging.info("No components found.")
-            return
-
-        self.supported_wine_runners = catalog["wine"]
-        self.supported_proton_runners = catalog["proton"]
-        self.supported_runtimes = catalog["runtimes"]
-        self.supported_winebridge = catalog["winebridge"]
-        self.supported_dxvk = catalog["dxvk"]
-        self.supported_vkd3d = catalog["vkd3d"]
-        self.supported_nvapi = catalog["nvapi"]
-        self.supported_latencyflex = catalog["latencyflex"]
+        def _run():
+            RepoStatus.repo_wait_operation("components.fetching")
+            RepoStatus.repo_start_operation("components.organizing")
+            catalog = self.component_manager.fetch_catalog()
+            if len(catalog) == 0:
+                logging.info("No components found.")
+                return
+
+            self.supported_wine_runners = catalog["wine"]
+            self.supported_proton_runners = catalog["proton"]
+            self.supported_runtimes = catalog["runtimes"]
+            self.supported_winebridge = catalog["winebridge"]
+            self.supported_dxvk = catalog["dxvk"]
+            self.supported_vkd3d = catalog["vkd3d"]
+            self.supported_nvapi = catalog["nvapi"]
+            self.supported_latencyflex = catalog["latencyflex"]
+            RepoStatus.repo_done_operation("components.organizing")
+        RunAsync(_run)
 
     def organize_dependencies(self):
```

Comment:
Could it be interesting to write a decorator to make this function run asynchronously without having to declare a function and do RunAsync? It could be used a lot in other places I think, and would make the code clearer in my opinion, but it could also make it too abstracted and prone to errors then.

What do you think?

Something like:
```python
@async
def organize_dependencies(self):
    """Organizes dependencies into supported_dependencies."""
    RepoStatus.repo_wait_operation("dependencies.fetching")
    ...

```

Reply:
I tried to make it many times without success